docs/design_docs.md

```diff
@@ -39,3 +39,126 @@ We need a shell, but how to represent it? We also need a virtual keyboard for on
 
 A web-GUI makes sense then with an HTTP interface to a REPL - `rappel` is a cool name ;-)
 
+## Overview
+
+The basic kick off document is here:
+
+https://medium.com/@gordonguthrie/the-beam-needs-an-apl-y-language-6c5c998ba6d
+
+## Rowan's thoughts 2020_05_04
+
+Heres an assortment of APL functions if you need testing targets.
+I tried to pick ones ranging from simple (mean) to deviously complex (trav).
+These were written for gnu apl, but nowadays dyalog is the most prominent implementation. There are subtle differences between the two I can help explain, or point you in the right direction for help.
+
+I have written them in the DFN style (see https://en.wikipedia.org/wiki/Direct_function). There is also tradfns (https://aplwiki.com/wiki/Tradfn).
```

Comment:
We will have both TradFns and DFNs - TradFns will correspond to Erlang and Elixir Fns and will be contained in modules (like Erlang Elixir) they will be declared private and public (like etc, etc).

The scope of TradFns will be function level (ie no global variables). DFNs will be closures over immutable values (like Funs in Erlang/Elixir).

I also intend that Pometo fns will be pure functions and side effect free so they will have to ability to call out to other languages or read data from external state.

The REPL in rappel will therefore be different to other APL REPLs.

It can be thought of as an Erlang function that calls out to Pometo ones.

You will be able to declare variables and bind them to the result of arbritary {module, function, argument} evals in Erlang/Elixir and then invoke your Pometo fns with those variables.

Rappel will be implemented in a gen server and the persistence of that gen servers state to the Binary Erlang Term Format means you will be able to persist and hydrate sessions, and email them to people.

We could potentially dump them to other formats that are more git friendly, lets see.